pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperStatusKitService.java

```diff
@@ -0,0 +1,149 @@
+        ResultSet foundKitResults = kitDao.getKitsByParticipantId(participantId);
+        return processNonPepperKitStatusFromResultSet(foundKitResults);
+    }
+
+    public KitResponse getKitsFromKitIds(String[] kitIdsArray) {
+        // get the kits with the given kit ids
+        try {
+            ArrayList<NonPepperKitStatus> list =  kitDao.getKitsByKitIdArray(kitIdsArray, this);
+            StatusKitResponse statusKitResponse = new StatusKitResponse(list);
+            return statusKitResponse;
+
+        } catch (Exception e) {
+            log.error("Error getting kits by an array of kit ids", e);
+            return new KitResponseError(KitResponse.DSM_ERROR);
+        }
+    }
+
+    private KitResponse processNonPepperKitStatusFromResultSet(ResultSet foundKitResults) {
+        ArrayList<NonPepperKitStatus> list = selectAllNonPepperKitStatus(foundKitResults);
+        StatusKitResponse statusKitResponse = new StatusKitResponse(list);
+        return statusKitResponse;
+    }
+
+    public ArrayList<NonPepperKitStatus> selectAllNonPepperKitStatus(ResultSet foundKitResults) {
+        HashMap<Integer, UserDto> users = UserDao.selectAllUsers();
+        ArrayList<NonPepperKitStatus> kits = new ArrayList<>();
+        try {
+            while (foundKitResults.next()) {
+                kits.add(createStatusForKit(foundKitResults, users));
+            }
+
+        } catch (SQLException e) {
+            log.error("Some Error getting the status of kits", e);
+        }
+        return kits;
+    }
+
+    private NonPepperKitStatus createStatusForKit(ResultSet foundKitResults, HashMap<Integer, UserDto> users) throws SQLException {
+        String labelUser = getUserEmailForFields(foundKitResults.getString(DBConstants.LABEL_BY), users);
+        String scannedUser = getUserEmailForFields(foundKitResults.getString(DBConstants.SCAN_BY), users);
+        String deactivatedUser = getUserEmailForFields(foundKitResults.getString(DBConstants.DEACTIVATED_BY), users);
+        String discardUser = getUserEmailForFields(foundKitResults.getString(DBConstants.DISCARD_BY), users);
+        String trackingScanBy = getUserEmailForFields(foundKitResults.getString(DBConstants.TRACKING_SCAN_BY), users);
+
+        String labelDate = convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.LABEL_DATE));
+        String scanDate = convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.DSM_SCAN_DATE));
+        String deactivatedDate = convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.DSM_DEACTIVATED_DATE));
+        String receivedDate = convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.DSM_RECEIVE_DATE));
+        String discardDate = convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.DISCARD_DATE));
+
+        NonPepperKitStatus nonPepperKitStatus =
+                new NonPepperKitStatus(foundKitResults.getString(DBConstants.DDP_KIT_REQUEST_ID),
```

Comment:
This is difficult to read and seems brittle. Some other choices: use a builder pattern, make the NonPepperKitStatus fields public and assign directly.
I realize the benefit of this approach is that every field is accounted for, but it's really difficult to read this code and determine that the current argument list is correct.